perl/toolbox/json.pm

```diff
@@ -98,35 +114,58 @@ sub put_json_file {
     my $json_ref = shift;
     my $schema_filename = shift;
     my $coder = JSON::XS->new->canonical->pretty;
-    my $errors = validate_schema($schema_filename, $filename, $json_ref);
-    if ($errors == 0) {
+    my $result = validate_schema($schema_filename, $filename, $json_ref);
+    if ($result == 0) {
         my $json_text = $coder->encode($json_ref);
         if (! defined $json_text) {
-            print "put_json_file(): JSON encode failed\n";
-            return 1;
+            # data json invalid
+            return 6;
         }
-        my $json_fh = open_write_text_file($filename);
-        if (defined $json_fh) {
+        (my $rc, my $json_fh) = open_write_text_file($filename);
+        if ($rc == 0 and defined $json_fh) {
             printf $json_fh "%s", $json_text;
             close($json_fh);
+            # all good
             return 0;
+        } elsif ($rc == 1) {
+            # data file name undefined
+            return 7;
+        } elsif ($rc == 3) {
+            # cannot open data file
+            return 9;
         } else {
-            print "put_json_file(): open_write_text_file() failed\n";
-            return 1;
+            # something else
+            return $rc;
         }
+    } elsif ($result == 1) {
+        # schema file name undefined
+        return 1
+    } elsif ($result == 2) {
+        # schema file not found
+        return 2;
+    } elsif ($result == 3) {
+        # cannot open schema file
+        return 3;
+    } elsif ($result == 4) {
+        # schema invalid
+        return 4;
+    } elsif ($result == 4) {
```

Comment:
`$result == 5`?

Reply:
any particular reason you wouldn't just do the following?

```
} else return $result;
```

for all cases where `$result != 0`

Reply:
see new code with debug logs